### DISCUSSION TITLE
Customizing OGP Settings for Private GitHub Pages in Organization Repository

### DISCUSSION THREAD

**Discussion by cc-harashima** [community/community#130852] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I'm deploying a GitHub Pages site with private visibility using a private repository within an organization. The content includes pages generated with Jekyll and OpenAPI specifications. The security is fine as only users within the GitHub Organization can access it.

However, when I post the link in our internal Slack, the OGP that appears includes the GitHub logo and the description: "GitHub is where people build software. More than 100 million people use GitHub to discover, fork, and contribute to over 420 million projects."

Is there any way to remove or adjust this?

**Comment by yoannchaudet** [comment]
Hello,

That sounds like a question for Slack instead. What you are seeing is probably the result of their classic unfurling: https://api.slack.com/reference/messaging/link-unfurling#classic_unfurl.

If your Private site is not using a custom domain, without access to the site itself, they may pull information from GitHub directly.

**Reply by cc-harashima** [reply]
Hello,

Thank you for the information.
I would like to clarify a few points. The private site is using a custom domain.
Additionally, for public GitHub Pages, the specified page's OG information is displayed correctly when posted on Slack.

Given that the private site can only be accessed by GitHub users who belong to our organization, it seems that the issue with the Classic Unfurl not retrieving the OG (meta information) of the relevant page when posted on Slack is due to access restrictions.
Indeed, there may be a way to customize the unfurling process and manage it via the API.

Thank you once again for your guidance.

**Reply by cc-harashima** [reply]
**Update:**
Attempting to access the private GitHub Pages using an API token appeared to be quite challenging: https://github.com/orgs/community/discussions/59659

**Reply by yoannchaudet** [reply]
Correct, GitHub Pages does not support API access today.

The recommended way would be to:

- Make sure your site is built with GitHub Actions (it's the case for most sites already)
- Keep around the Actions artifact and pull the information you need from there

**Reply by cc-harashima** [reply]
GitHub Pages is built using GitHub Actions.
The issue at hand is not about the method of building and deploying GitHub Pages, but rather the fact that, when setting a custom domain on private GitHub Pages, the OGP and meta information can only be set by GitHub.
This issue is demonstrated in Slack, but it is likely to occur when posting to Twitter or Facebook as well.

Ideally, we would like to configure it as follows, although it may not be possible at the moment:
* Remove OGP or meta information
* Change OGP or meta information to fixed ones
* Allow only OGP or meta information to pass through the information of GitHub Pages

If no solution can be found, it might be better to host it in another environment instead of using GitHub Pages.

**Reply by yoannchaudet** [reply]
Custom domain or not, when your site is private, any crawler trying to access it will receive a HTTP redirect to the GitHub login page. That's where Slack's default unfurling is probably getting the information you are seeing.

I am afraid if you need better unfurling support in say, Slack, you will need to write a Slack app for that purpose.

**Reply by cc-harashima** [reply]
Yes, it does seem that the OG for the login screen is being displayed.
This time, I will use the Slack app to control the unfurl for the target domain.

Thank you for your cooperation.
